dask/dataframe/io/parquet/arrow.py

```diff
@@ -454,9 +454,13 @@ def extract_filesystem(
                 urlpath = [stringify_path(urlpath)]
 
             if fs in ("arrow", "pyarrow"):
-                fs = type(pa_fs.FileSystem.from_uri(urlpath[0])[0])(
-                    **(storage_options or {})
-                )
+                fs = pa_fs.FileSystem.from_uri(urlpath[0])[0]
+                if storage_options:
+                    # Use inferred region as the default
+                    region = (
+                        {} if "region" in storage_options else {"region": fs.region}
+                    )
+                    fs = type(fs)(**region, **storage_options)
```

Comment:
Maybe it helps here, one can specify region as part of the URI:
`s3://[access_key:secret_key@]bucket/path[?region=]`
https://arrow.apache.org/docs/r/articles/fs.html#connecting-directly-with-a-uri

Reply:
I suppose storage options can also contain a token, and that I can't find anything about including the token in the URI; don't let this comment hinder merging though.

Reply:
Down the rabbit hole a bit, doesn't appear one can set access token in the URI; though query parameters are region, scheme, endpoint_override, and some others:
https://github.com/apache/arrow/blob/6e6fd0a30aebee6738bb4bb703344f0c866debea/cpp/src/arrow/filesystem/s3fs.cc#L334

Reply:
> Maybe it helps here, one can specify region as part of the URI: s3://[access_key:secret_key@]bucket/path[?region=]

Just to make sure we're all on the same page, this is a user-facing API.  I expect that 99% of users won't do what's suggested in those docs, so that unfortunately doesn't help us.  It's important that we include the user when thinking about our design.

Reply:
The user's interaction wasn't my intent here. The topic came up if we could avoid re-initializing the FileSystem. As such, if the URI format supported everything the storage_options may contain, we could transparently pass it thru in the URI was all. (which it turns out it doesn't)

Reply:
Thanks for looking into this @milesgranger !

If the region cannot be correctly inferred for some reason, then the user would be able to specify a specific region within the URI **or** within `storage_options` after this PR. So hopefully this makes things relatively foolproof.

I agree with Matt that is should be okay to re-initialize the FileSystem object. However, if it makes you feel any better: When the user configures their credentials with something like aws-cli, they shouldn't need to pass in `storage_options`, and so we won't need to re-initialize the FileSystem object. (Disclaimer: My personal credentials are expired, so I haven't verified this yet)